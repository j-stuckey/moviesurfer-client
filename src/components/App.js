import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import LandingPage from './LandingPage';
import Search from './Search';
import Login from './Login';
import PrivateRoute from './PrivateRoute';
import { NoMatch } from './NoMatch';
import { Header } from './Header';
import RegisterPage from './RegisterPage';

import PropTypes from 'prop-types';

export class App extends Component {
    render() {
        return (
            <Router>
                <React.Fragment>
                    <Route path="/" component={Header}/>
                    <Switch>
                        <Route exact path="/" component={LandingPage} />
                        <Route exact path="/login" component={Login} />
                        <Route exact path="/register" component={RegisterPage} />
                        <PrivateRoute
                            path="/search"
                            isLoggedIn={this.props.isLoggedIn}
                            component={Search}
                        />
                        <Route component={NoMatch} />
                    </Switch>
                </React.Fragment>
            </Router>
        );
    }
}

App.propTypes = {
    isLoggedIn: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
    hasAuthToken: state.auth.authToken !== null,
    isLoggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(App);
