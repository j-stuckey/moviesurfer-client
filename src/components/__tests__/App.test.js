import React from 'react';
import { App } from '../App';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

describe('<App />', () => {
    const wrapper = shallow(<App />);

    it('renders without crashing', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
