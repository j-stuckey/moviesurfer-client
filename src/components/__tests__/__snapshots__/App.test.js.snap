// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      path="/search"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
